### ISSUE TITLE
Test Planemo Bus Factor

### ISSUE THREAD

**Issue by jmchilton** [galaxyproject/planemo#329]
Have someone besides @jmchilton do a release of planemo.

Update  - current plan is
- [ ] Implement release testing #184.
- [ ] Modify `make release` to test release instead of reviewing testpypi.
- [ ] Modify `make release` to infer default versions automatically.
- [ ] Update galaxyproject's jenkins to be able to publish to PyPI.
- [ ] Add a jenkins job to release Planemo.
- [ ] Have @davebx put together release notes and do a release.

**Comment by erasche** [comment]
Oooh fun. I vote for jenkins :wink: 

Continuous Delivery ftw.

**Comment by davebx** [comment]
A wild volunteer appeared!

**Comment by erasche** [comment]
@davebx ooh yay a volunteer! See if you can't checklist-ify and automate https://github.com/galaxyproject/planemo/blob/master/docs/developing.rst :)

**Comment by jmchilton** [comment]
@davebx - How about you push the button but we configure jenkins with the keys.

We should I guess fully automate:

```
make clean && make lint && make test
make release VERSION=<old_version> NEW_VERSION=<new_version>
make push-release
```

There is a review step in there, it pauses and opens a web browser to Test PyPI but maybe that is not needed. Maybe the right thing is to just do an install and run some simple tests.

**Comment by jmchilton** [comment]
xref https://github.com/galaxyproject/planemo/issues/184

If that worked we could do a release to test PyPI and not need an explicit review I think.

**Comment by erasche** [comment]
:100:

**Comment by jmchilton** [comment]
P.S. I'm not planning on quiting or getting hit by a bus, but I realized I totally merged #327 when other people when doing just fine working on the issue - it would be better for me to step back and let the project evolve from under my heavy boot :smile:.

**Comment by davebx** [comment]
@jmchilton Yes, too long has planemo languished in the iron fist of its tyrant lord. Or something.

**Comment by erasche** [comment]
(P.S. I was actually hoping you'd sign off/sanity check that one since you wrote the vast majority of the code surrounding it/affected by it.)

**Comment by jmchilton** [comment]
Updated this issue with a checklist based on these conversations - feel free to tweak.

#184 seemed like a cool but non-essential idea at one point but now seems more and more important to being able to have automated, stable releases.

**Comment by erasche** [comment]
@davebx how goes the release process? :)

**Comment by jmchilton** [comment]
@erasche I've decided to retain sole responsibility for releases until new checklist is complete, I'll do a release now if you wish.

**Comment by erasche** [comment]
@jmchilton oh, okay, sounds good! No, release whenever you feel is appropriate. I imagine many of us (jenkins bot included) are using `:master` and not too concerned about the public release process. Just wanted to follow up on this :)

**Comment by jmchilton** [comment]
Had an explicit request for an update from @kellrott so pushed 0.18.0.

**Comment by peterjc** [comment]
@erasche I'm using ``pip install planemo`` within my ``.travis.yml`` setups, rather than working with the master branch - so when I want to use something new I have occasionally waited until an official release came out.

**Comment by erasche** [comment]
@peterjc okay, good to know, thanks for the data point!

**Comment by jmchilton** [comment]
Multiple people have published this now - bus factor up!
